### PR TITLE
[online-event] Fix typing on fee level

### PR DESCRIPTION
**Pull request by eileenmcnaughton**
Overview
----------------------------------------
Fix typing on fee level

Before
----------------------------------------
image

After
----------------------------------------
poof

Technical Details
----------------------------------------
I pushed the wrangling to the only calling functions & fixed in the functions that call that

Comments
----------------------------------------

### CODE REVIEW

**Comment by eileenmcnaughton** [comment]
test this please

**Comment by eileenmcnaughton** [comment]
@colemanw one more minor event online fix to merge if you can (I'm doing some testing now so good to have all the fixes in)

File: CRM/Contribute/BAO/Contribution.php [hunk 1]
```diff
@@ -3387,6 +3387,7 @@ public static function recordFinancialAccounts(&$params, CRM_Contribute_DAO_Cont
     ) {
       $eventID = CRM_Core_DAO::getFieldValue('CRM_Event_DAO_Participant', $entityId, 'event_id');
       $feeLevel[] = str_replace('', '', $params['prevContribution']->amount_level);
+      $feeLevel = (string) current($feeLevel);
```
Comment (colemanw) [inline CRM/Contribute/BAO/Contribution.php:3390]: These two lines seem like a non-sequitur.  First it adds an item to an array, then overwrites the array with that item? Why not just
```suggestion
      $feeLevel = str_replace('', '', $params['prevContribution']->amount_level);
```

Comment (eileenmcnaughton) [reply CRM/Contribute/BAO/Contribution.php]: Yeah I think I focussed on moving the code to the right place - but sometimes when you do that the next step becomes clear!

**Pull request merged**